src/tests/ftest/control/dmg_network_scan.py

```diff
@@ -4,273 +4,72 @@
-            elif cfg_p == "ofi+verbs":
-                cfg_p = "ofi+verbs;ofi_rxm"
-            f_net_devs = [dev for dev in f_net_devs if dev.providers == [cfg_p]]
-
-        return f_net_devs
+        server_provider = self.server_managers[0].get_config_value("provider")
+        sys_info = []
+        for entry in get_network_information(self.hostlist_servers, SUPPORTED_PROVIDERS):
+            if entry.device.startswith("ib") and server_provider in entry.provider:
+                sys_info.append(entry)
+                sys_info[-1].ib_device = None
+        return sys_info
 
     def get_dmg_info(self):
-        """Store the information received from dmg output.
+        """Get the information received from dmg network scan output.
 
         Returns:
-            list: list of NetDev objects.
+            list: list of NetworkDevice objects.
 
         """
-        host = None
-        numa = None
-        temp_net_devs = []
-        dmg_net_devs = []
-
-        # Get dmg info
-        network_out = self.dmg.network_scan()
-
-        # Get devices divided into dictionaries, with iface being the key. e.g.,
-        # temp_net_devs =
-        # [
-        #     {
-        #         'ib0': [['ofi+sockets'], 'wolf-154', '0'],
-        #         'ib1': [['ofi+sockets'], 'wolf-154', '1'],
-        #         'eth0': [['ofi+sockets'], 'wolf-154', '0']
-        #     }
-        # ]
-        host_fabrics = network_out["response"]["HostFabrics"]
-        struct_hash = list(host_fabrics.keys())[0]
-        interfaces = host_fabrics[struct_hash]["HostFabric"]["Interfaces"]
-
-        # Fill in the dictionary.
-        parsed_devs = {}
-        for interface in interfaces:
-            device = interface["Device"]
-            provider = interface["Provider"]
-            if device in parsed_devs:
-                parsed_devs[device][0].append(provider)
-            else:
-                host = host_fabrics[struct_hash]["HostSet"].split(":")[0]
-                parsed_devs[device] = [
-                    [provider], host, str(interface["NumaNode"])]
-
-        temp_net_devs.append(parsed_devs)
-
-        # Create NetDev objects
-        for d in temp_net_devs:
-            for iface in d:
-                net_dev = [NetDev(d[iface][1], iface, [provider], d[iface][2])
-                           for provider in d[iface][0]]
-                dmg_net_devs.extend(net_dev)
-
-        return dmg_net_devs
+        dmg = self.get_dmg_command()
+        return get_dmg_network_information(dmg.network_scan())
 
     def test_dmg_network_scan_basic(self):
-        """
-        JIRA ID: DAOS-2516
+        """JIRA ID: DAOS-2516
+
         Test Description: Test basic dmg functionality to scan the network
         devices on the system.
-        :avocado: tags=all,small,daily_regression,hw,dmg,network_scan,basic
+
+        :avocado: tags=all,daily_regression
+        :avocado: tags=hw,small
+        :avocado: tags=dmg,network_scan,basic
```

Comment:
We've been slowly adding the `control` feature

```suggestion
        :avocado: tags=all,daily_regression
        :avocado: tags=hw,small
        :avocado: tags=dmg,control
        :avocado: tags=network_scan,basic
```

Reply:
Done